src/halmos/sevm.py

```diff
@@ -1047,8 +1044,7 @@ def is_jumpdest(self, x: Word) -> bool:
         if pc < 0:
             raise ValueError(pc)
 
-        opcode = unbox_int(self.pgm[pc])
-        return opcode == EVM.JUMPDEST
+        return pc in self.pgm.valid_jump_destinations()
```

Comment:
i don't think it is correct.  because of the stopping behavior of CodeIterator, `valid_jump_destinations()` may not include all jump destinations, i.e., the ones after a symbolic byte.  e.g., immutable storage variables can be initialized with a symbolic value, then the symbol appear in the bytecode, and then CodeIterator stops at the immutable symbol, so any jumpdest opcode following the symbol will not be included in the valid_jump_destinations set.

Reply:
i was wrong. the iterator iterates over instructions (that consist of opcode and arguments if any), not over bytes. since the symbolic immutable values appear as an argument of the push opcode, the iterator will not stop prematurely there.

but there is still a case where this change makes differences. if there is a symbolic opcode in the middle of bytecode that is not reachable (deadcode), then the iterator approach could miss the reachable part of bytecode after the deadcode. this seems to be a fundamental question how to precisely compute valid jumpdests in the presence of possibly symbolic deadcode. since this is an orthogonal topic, and it is not likely to happen in most practical cases, we can merge this pr for now, and have a separate discussion (probably in #251 and #252).

Reply:
true, in all the cases I considered the data-in-bytecode was appended at the end:

- metadata
- constructor arguments in init code
- jump table values

So these would not cause an issue, but in principle it's certainly possible to have a jump table at the beginning of bytecode. Maybe to handle this we could consider that symbolic values in bytecode implicitly are data, and we could resume jumpdest processing when concrete data resumes?

Reply:
That would be an option, but if the data section consists of both symbolic and concrete values, then it wouldn't be trivial to determine when resuming jumpdest processing. Perhaps we can move this discussion to #252 ?